docarray/array/array/sequence_indexing_mixin.py

```diff
@@ -0,0 +1,216 @@
+        is_valid_bulk_index = index_has_getitem and isinstance(item, Iterable)
+        if not is_valid_bulk_index:
+            raise ValueError(f'Invalid index type {type(item)}')
+
+        if isinstance(item, np.ndarray) and (
+            item.dtype == np.bool_ or np.issubdtype(item.dtype, np.integer)
+        ):
+            return item.tolist()
+
+        # torch index types
+        torch_available = is_torch_available()
+        if torch_available:
+            import torch
+        else:
+            raise ValueError(f'Invalid index type {type(item)}')
+        allowed_torch_dtypes = [
+            torch.bool,
+            torch.int64,
+        ]
+        if isinstance(item, torch.Tensor) and (item.dtype in allowed_torch_dtypes):
+            return item.tolist()
+
+        return item
+
+    def _get_from_indices(self: T, item: Iterable[int]) -> T:
+        results = []
+        for ix in item:
+            results.append(self._data[ix])
+        return self.__class__(results)
+
+    def _set_by_indices(self: T, item: Iterable[int], value: Iterable[T_item]):
+        # here we cannot use _get_offset_to_doc() because we need to change the doc
+        # that a given offset points to, not just retrieve it.
+        # Future optimization idea: _data could be List[DocContainer], where
+        # DocContainer points to the doc. Then we could use _get_offset_to_container()
+        # to swap the doc in the container.
+        for ix, doc_to_set in zip(item, value):
+            try:
+                self._data[ix] = doc_to_set
+            except KeyError:
+                raise IndexError(f'Index {ix} is out of range')
+
+    def _get_from_mask(self: T, item: Iterable[bool]) -> T:
+        return self.__class__(
+            [doc for doc, mask_value in zip(self, item) if mask_value]
+        )
+
+    def _set_by_mask(self: T, item: Iterable[bool], value: Sequence[T_item]):
+        i_value = 0
+        for i, mask_value in zip(range(len(self)), item):
+            if mask_value:
+                self._data[i] = value[i_value]
+                i_value += 1
+
+    def _del_from_mask(self: T, item: Iterable[bool]) -> None:
+        idx_to_delete = [i for i, val in enumerate(item) if val]
+        self._del_from_indices(idx_to_delete)
+
+    def _del_from_indices(self: T, item: Iterable[int]) -> None:
+        for ix in sorted(item, reverse=True):
+            # reversed in needed here otherwise some the indice are not up to date after
+            # each delete
+            del self._data[ix]
+
+    def __delitem__(self, key: Union[int, IndexIterType]) -> None:
+        item = self._normalize_index_item(key)
+
+        if item is None:
+            return
+        elif isinstance(item, (int, slice)):
+            del self._data[item]
+        else:
+            head = item[0]  # type: ignore
+            if isinstance(head, bool):
+                item_ = cast(Iterable[bool], item)
+                return self._del_from_mask(item_)
```

Comment:
Here it looks like can actually do the more advanced stuff for deletion as well. So is the docstring outdated?